lib/Basics/MemoryTypes/MemoryTypes.h

```diff
@@ -0,0 +1,69 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Heiko Kernbach
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Basics/ResourceUsage.h"
+
+#include <string>
+#include <unordered_map>
+
+namespace arangodb {
+typedef std::basic_string<char, std::char_traits<char>,
+                          ResourceUsageAllocator<char, ResourceMonitor>>
+    MonitoredString;
+
+struct hash_monitored_string {
+  using is_transparent = void;
+  [[nodiscard]] size_t operator()(const char* txt) const {
```

Comment:
Our convention is to do it like this:
```suggestion
  [[nodiscard]] size_t operator()(char const* txt) const {
```